### PR TITLE
i#1556: Mark x86 rep instructions as predicated.

### PR DESCRIPTION
**Pull request by ivankyluk**
i#1556: Mark x86 rep instructions as predicated.

Issue: #1556

### CODE REVIEW

File: core/ir/x86/decode_table.c [hunk 1]
```diff
@@ -9597,33 +9597,33 @@ const instr_info_t repne_extensions[][6] = {
...
   },
   { /* repne extension 1 */
     {OP_cmps,       0xa70000, "cmps",         axSI, axDI, Xv, Yv, axSI, xop_next, (fW6|fRD), tne[0][0]},
     {OP_CONTD,      0xa70000, "cmps",         xx, xx, axDI, xx, xx, no, (fW6|fRD), END_LIST},
-    {OP_rep_cmps,   0xf3a70000, "rep cmps",   axSI, axDI, Xv, Yv, axSI, xop_next, (fW6|fRD|fRZ), tne[0][2]},
-    {OP_CONTD,      0xf3a70000, "rep cmps",   axCX, xx, axDI, axCX, xx, no, (fW6|fRD), END_LIST},
-    {OP_repne_cmps, 0xf2a70000, "repne cmps", axSI, axDI, Xv, Yv, axSI, xop_next, (fW6|fRD|fRZ), tne[0][4]},
-    {OP_CONTD,      0xf2a70000, "repne cmps", axCX, xx, axDI, axCX, xx, no, (fW6|fRD), END_LIST},
+    {OP_rep_cmps,   0xf3a70000, "rep cmps",   axSI, axDI, Xv, Yv, axSI, xop_next|predcx, (fW6|fRD|fRZ), tne[0][2]},
+    {OP_CONTD,      0xf3a70000, "rep cmps",   axCX, xx, axDI, axCX, xx, predcx, (fW6|fRD), END_LIST},
```
Comment (ivankyluk) [inline core/ir/x86/decode_table.c:9608]: Copy the comment from https://github.com/DynamoRIO/dynamorio/pull/6283

> Hmm this is missing fRZ -- ditto for the others. Does the decoder take the flags from the final entry? I think it does.

I might have missed it.

read_instruction in ir/x86/decode.c 

    } else if (info->type == REP_EXT) {
        /* discard old info, get new one */
        int code = (int)info->code;
        int idx = (di->rep_prefix ? 2 : 0);
        info = &rep_extensions[code][idx];
        if (di->rep_prefix)
            di->rep_prefix = false;

points to the first entry,  and decode_opcopde/decode_common sets eflags after getting the entry by calling read_instruction.

So, it appears the eflag from the first entry is used.

Comment (derekbruening) [reply core/ir/x86/decode_table.c:9608]: Still seems best to give them the same values (could even have decoder check in debug build).

Comment (ivankyluk) [reply core/ir/x86/decode_table.c]: I've updated the table accordingly, and modified the test to cover eflags.

Comment (ivankyluk) [reply core/ir/x86/decode_table.c]: PTAL.

**Review by derekbruening — APPROVED** [review]

**Pull request merged**